src/Blocks/BlockConcretePowder.h

```diff
@@ -0,0 +1,98 @@
+
+#pragma once
+
+#include "BlockHandler.h"
+
+
+
+
+
+class cBlockConcretePowderHandler :
+	public cBlockHandler
+{
+public:
+	cBlockConcretePowderHandler(BLOCKTYPE a_BlockType)
+		: cBlockHandler(a_BlockType)
+	{
+	}
+
+
+
+
+
+	virtual void Check(cChunkInterface & a_ChunkInterface, cBlockPluginInterface & a_PluginInterface, int a_RelX, int a_RelY, int a_RelZ, cChunk & a_Chunk) override
+	{
+		GetSoaked(Vector3i(a_RelX, a_RelY, a_RelZ), a_Chunk);
+	}
+
+
+
+
+
+	/** Check blocks above and around to see if they are water. If one is, convert this into concrete block */
+	void GetSoaked(Vector3i a_Rel, cChunk & a_Chunk)
+	{
+		bool ShouldSoak = false;
+
+		static const Vector3i Coords[] =
+		{
+			Vector3i( 1, 0,  0),
+			Vector3i(-1, 0,  0),
+			Vector3i( 0, 0,  1),
+			Vector3i( 0, 0, -1),
+			Vector3i( 0, 1,  0),
+		};
+		for (size_t i = 0; i < ARRAYCOUNT(Coords); i++)
+		{
+			if (IsBlockWater(a_Chunk.GetBlock({a_Rel.x + Coords[i].x, a_Rel.y + Coords[i].y, a_Rel.z + Coords[i].z})))
```

Comment:
This has to use `a_Chunk.UnboundedRelGetBlockType` because the neighbours might be in a different chunk.